syft/core/workers.py

```diff
@@ -60,8 +60,9 @@ class BaseWorker(object):
           print events to stdout.
 
     """
+
     def __init__(self,  hook=None, id=0, is_client_worker=False, objects={},
-                 tmp_objects={}, known_workers={}, verbose=True):
+                 tmp_objects={}, known_workers={}, verbose=True, queue_size=0):
```

Comment:
I'm not sure we want to initialize workers with queue_size.  There are times where sending commands in batches is helpful for speed, and there are times where I want to be interactively querying remote workers, likely all in the same script.  The way it's written now, I'd have to run two client workers to be able to do that.  We spoke a bit about including this as a parameter in the context manager, so should we be putting it in TorchHook?

Reply:
If you look at the behavior of send message, if queue size is 0 ( the default) messages are sent exactly as they were before. Torch hook also needs a parameter for this. Thank you for reminding me

Reply:
But if this is static in the worker, wouldn't I have to reinitialize the worker or create a new worker to change the queue size?

Reply:
One planned future improvement is to allow for the modification of that code at runtime with a context manager/setter which changes the size

Reply:
Could it be dynamic with the number of owners of a tensor? It seems highly likely there will be tensors with varying numbers of owners.

Reply:
Will add issue to make this parameter able to be a callable to take care of all of these options. In practice for the beginning it will either be size 0 or infinity